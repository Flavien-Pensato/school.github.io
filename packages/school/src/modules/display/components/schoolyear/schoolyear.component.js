import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

export class SchoolYear extends Component {
  constructor(props) {
    super(props);

    this.state = {
      schoolyears: this.generateSchoolyears(),
    };
  }

  generateSchoolyears = () => {
    const schoolyears = {};
    const date = moment().add(-2, 'year');

    for (let len = 0; len < 4; len += 1) {
      schoolyears[`${date.year()}-${date.add(1, 'year').year()}`] = true;
    }

    return schoolyears;
  };

  render() {
    const { current } = this.props;

    return (
      <div>
        <select>
          {Object.keys(this.state.schoolyears).map(year =>
            <option key={year}value={year} selected={year === current}>{year}</option>)}
        </select>
      </div>
    );
  }
}

SchoolYear.propTypes = {
  current: PropTypes.string.isRequired,
};
