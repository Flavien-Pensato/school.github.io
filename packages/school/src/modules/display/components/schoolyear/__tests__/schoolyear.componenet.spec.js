import React from 'react';
import { shallow } from 'enzyme';
import { SchoolYear } from '../schoolyear.component';

describe('<SchoolYear />', () => {
  const props = {};
  const getWrapper = () => shallow(<SchoolYear {...props} />);

  beforeEach(() => {
    props.current = '9';
  });

  it('should render normally', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
