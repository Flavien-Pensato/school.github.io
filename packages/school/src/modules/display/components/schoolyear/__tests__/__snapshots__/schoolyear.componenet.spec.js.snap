// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SchoolYear /> should render normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SchoolYear
    current="9"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select>
        <option
          selected={false}
          value="2016-2017"
        >
          2016-2017
        </option>
        <option
          selected={false}
          value="2017-2018"
        >
          2017-2018
        </option>
        <option
          selected={false}
          value="2018-2019"
        >
          2018-2019
        </option>
        <option
          selected={false}
          value="2019-2020"
        >
          2019-2020
        </option>
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            selected={false}
            value="2016-2017"
          >
            2016-2017
          </option>,
          <option
            selected={false}
            value="2017-2018"
          >
            2017-2018
          </option>,
          <option
            selected={false}
            value="2018-2019"
          >
            2018-2019
          </option>,
          <option
            selected={false}
            value="2019-2020"
          >
            2019-2020
          </option>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2016-2017",
          "nodeType": "host",
          "props": Object {
            "children": "2016-2017",
            "selected": false,
            "value": "2016-2017",
          },
          "ref": null,
          "rendered": "2016-2017",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2017-2018",
          "nodeType": "host",
          "props": Object {
            "children": "2017-2018",
            "selected": false,
            "value": "2017-2018",
          },
          "ref": null,
          "rendered": "2017-2018",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2018-2019",
          "nodeType": "host",
          "props": Object {
            "children": "2018-2019",
            "selected": false,
            "value": "2018-2019",
          },
          "ref": null,
          "rendered": "2018-2019",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2019-2020",
          "nodeType": "host",
          "props": Object {
            "children": "2019-2020",
            "selected": false,
            "value": "2019-2020",
          },
          "ref": null,
          "rendered": "2019-2020",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select>
          <option
            selected={false}
            value="2016-2017"
          >
            2016-2017
          </option>
          <option
            selected={false}
            value="2017-2018"
          >
            2017-2018
          </option>
          <option
            selected={false}
            value="2018-2019"
          >
            2018-2019
          </option>
          <option
            selected={false}
            value="2019-2020"
          >
            2019-2020
          </option>
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected={false}
              value="2016-2017"
            >
              2016-2017
            </option>,
            <option
              selected={false}
              value="2017-2018"
            >
              2017-2018
            </option>,
            <option
              selected={false}
              value="2018-2019"
            >
              2018-2019
            </option>,
            <option
              selected={false}
              value="2019-2020"
            >
              2019-2020
            </option>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2016-2017",
            "nodeType": "host",
            "props": Object {
              "children": "2016-2017",
              "selected": false,
              "value": "2016-2017",
            },
            "ref": null,
            "rendered": "2016-2017",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2017-2018",
            "nodeType": "host",
            "props": Object {
              "children": "2017-2018",
              "selected": false,
              "value": "2017-2018",
            },
            "ref": null,
            "rendered": "2017-2018",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2018-2019",
            "nodeType": "host",
            "props": Object {
              "children": "2018-2019",
              "selected": false,
              "value": "2018-2019",
            },
            "ref": null,
            "rendered": "2018-2019",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2019-2020",
            "nodeType": "host",
            "props": Object {
              "children": "2019-2020",
              "selected": false,
              "value": "2019-2020",
            },
            "ref": null,
            "rendered": "2019-2020",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
